### PR TITLE
[Reviewer: G] Add better diagnostics to help track down https://github.com/Metaswit…

### PR DESCRIPTION
**Pull request by rkd-msw**
…ch/memento/issues/136

### CODE REVIEW

File: lib/tests/memento.rb [hunk 1]
```diff
@@ -60,6 +60,8 @@ def check_users_call_list user, from, to, answered, caller_id = 0
   # Check some fields are as expected. First the From URI.
   if caller_id == ANONYMOUS_URI
     fail "Call record doesn't contain anonmyised from\n#{call.xml}" unless call.from_uri == "sip:anonymous@anonymous.invalid"
+  elsif call.nil?
+    fail "No calls in call record - original XML was:\n#{call_list.original_xml}"
```
Comment (graemerobertson) [inline lib/tests/memento.rb]: This is a pretty random place to put this check? And if we go through the branch above (i.e. `caller_id == ANONYMOUS_URI`) we'll hit an exception there.

Let's check the length of `call_list` straight after the call to `get_call_list`

**Comment by graemerobertson** [comment]
Looks basically terrible to me but maybe I've missed something

**Comment by graemerobertson** [comment]
:dromedary_camel:

**Comment by rkd-msw** [comment]
Cool.

**Pull request merged**